### ISSUE TITLE
Daylight-color-mode

### ISSUE THREAD

**Issue by ratmalwer** [Stellarium/stellarium#678]
Today I was out in the field with my notebook to look what objects could be seen at night. Unfortunately the sun was shining bright. So I could not see anything on my screen (Stars, compass point or horizon). Even I turned the display on maximum brightnes.

Could it be possible to make a daylight-mode? I suggest inverse colors or better just black on white. This should affect the fullscreen including stars and texts. Preferably even the menues in an oldfashioned way without opacy.

Kind regards 
Markus

**Comment by axd1967** [comment]
Totaly agree on this issue, and seeing this tagged "opinion" and "wont fix" almost stopped my heart and has started wondering what's happening here. I almost missed the "PR accepted" note...

personally I'm struggling a lot with this theme, and I'm afraid I'm not alone. my eyes have lived over 50 years and are a bit worn...

the dark grey theme is a major UI issue . Take the astro calc window as an example: the graphs are almost unreadable (especially the thin red line on grey background, I would put my hand in fire that a UI expert would stumble over this, and I wonder how the color-blind percieve this). 

![image](https://user-images.githubusercontent.com/3529789/63294627-1c45d900-c2cb-11e9-91c2-00a1c57148e5.png)

 and one has to look very hard to see those see extra, faint lines: I've inverted the colors in an attempt to bring thise out:


![image](https://user-images.githubusercontent.com/3529789/63295191-5663aa80-c2cc-11e9-9d19-16b9a3f830ca.png)

![image](https://user-images.githubusercontent.com/3529789/63295273-8743df80-c2cc-11e9-956e-28bf9092798f.png)

Probably they were chosen to operate optimally in darkened conditions, which is understandable.

But those graphs are inherently meant to be used for planning purposes - i.e. during bright daylight.

I guess this a Qt theme that should become an option/setting; someone wanting to consult such graphs in dark condition can always enable night vision mode instead. 

Which, oddly, currently (0.19) provides a better viewing experience...

![image](https://user-images.githubusercontent.com/3529789/63294623-14863480-c2cb-11e9-9d01-1cca60099d2f.png)

 This should not be tagged "won"t fix" because quite rude, but, for example, "too expensive for the moment". and/or it could be put in a backlog pipeline, for example.

(and unfortunately unrelated, but the tooltips try to be helpful but fade away ar too quickly!)

**Comment by alex-w** [comment]
@axd1967 Are you go at the bright midday on the street for planning observation? Seriously? Are you just do not like this theme? No problem - just create a new theme (with code for selecting the color themes) and propose the pull request.

> and unfortunately unrelated, but the tooltips try to be helpful but fade away ar too quickly!

Seriously? When I put the mouse cursor on some UI element with tooltip then I got the tooltip and I see this tooltip all time until the cursor is staying over this element.

**Comment by axd1967** [comment]
* hello @alex-w - I use Stellarium in normal daylight for planning and I have trouble with the grey dialogs. I think I remember vagely that it was possible for a user to change a scheme without touching a Qt app. I need to check this. https://askubuntu.com/questions/909075/how-to-customize-the-colors-of-qt-applications-in-ubuntu-16-10/909216 and https://bbs.archlinux.org/viewtopic.php?id=149825
* About the tooltip: that is interesting, then there must be a problem on my side then: I run Ubuntu 16

**Comment by ratmalwer** [comment]
@alex-w 
Yes planning observation at daylight makes sense. As I live in an area where are mountains and the paths are not allways in good condition. I try to find the right spot at daylight, so I do not struggle in the dark.
As a photographer I try to find the perfect location this way and can start my timelapse before the object is rising at the horizon.

**Comment by axd1967** [comment]
I tried a bogus command `stellarium -stylesheet axd` and I get `QCss::Parser - Failed to load file  "axd"`. when I create a file and feed its name as argument, the warning disappears

`qt4config` launches fine and config can be adapted, but that does not seem to impact Stellarium.

I also hacked a bit, just to see if it would change anything without having to rebuild stellarium:

```patch
Index: plugins/RemoteControl/webroot/style.css
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- plugins/RemoteControl/webroot/style.css	(date 1553417612000)
+++ plugins/RemoteControl/webroot/style.css	(date 1566314753860)
@@ -82,7 +82,7 @@
 }
 
 .ui-tabs .ui-tabs-panel {
-    color: rgb(3,3,3);
+    color: rgb(238, 255, 15);
     background: linear-gradient(rgb(86, 87, 90),rgb(48, 49, 52));
     border-width: 0px;
 	border-radius: 0px;
@@ -202,7 +202,7 @@
 }
 
 button {
-	color: rgb(0, 0, 0);
+	color: rgb(255, 12, 51);
 }
 
 h1 {
@@ -530,7 +530,7 @@
 }
 
 ul.buttonlist > li > button:active{
-	background-color: rgb(180, 180, 183);
+	background-color: rgb(255, 189, 140);
 	border-style: inset;
 }
 
@@ -556,7 +556,7 @@
 
 ul.button32list > li > button:active,
 button.button32:active {
-	background-color: rgb(180, 180, 183);
+	background-color: rgb(25, 250, 60);
 }
 
 .icon32 {

```

**Comment by axd1967** [comment]
(I just realise that   it's funny to label this as [opinion](https://github.com/Stellarium/stellarium/labels/opinion) because strictly speaking the developer also has an opinion on that :-)

**Comment by alex-w** [comment]
@axd1967 you need to edit normalStyle.css file (https://github.com/Stellarium/stellarium/blob/master/data/gui/normalStyle.css) for creating a new theme and this is not possible without building Stellarium from source code. Of course adding new theme (+new css file for it) requires adding C++ code for selecting of the theme.

**Comment by axd1967** [comment]
thx, I've coded in Qt and C++ years ago. what i had in mind was something along `QT_STYLE_OVERRIDE` and `qt-config`. however, i'v never seen this mechansm in action, and google tells me it might not work as I imagine. I installed and tried qtconfig and noticed changes on Ubuntu 16, so I decided to backtrack a bit...

**Comment by axd1967** [comment]
another example of where my opinion tells me the color scheme is really bad:

![image](https://user-images.githubusercontent.com/3529789/64275795-b900ab80-cf46-11e9-9a4e-3cbc9190b96e.png)


I'm sorry, I don't want to give negative critic, but the marked text is unreadable to me.

**Comment by alex-w** [comment]
@axd1967 sorry, but it annoys me already. If our color theme is very bad, unreadable, invisible, etc. then where is your patch with good color theme, who can be good readable and visible in all cases and in all color spaces?

**Comment by axd1967** [comment]
hello @alex-w , upon rereading this issue, I think I dislodged a misunderstanding.

I don't ask for a color scheme that is visible in all circumstances, because there are at least two use cases that ask for totally different color schemes: dark room **and daylight** use cases. This issue cannot be solved with a single color scheme.

The current UI mode of Stellarium is a "dark room" mode (optionally enhanced with the "night mode" option), _but Stellarium contains an exception_, read on.

**A solution is to let the graphs be shown in daylight mode, and if those graphs must be shown in a dark room situation, the user switches on "night mode".**

To illustrate this I take the current situation, where the date/time dialog is shown to simulate a "daylight mode" dialog (all other dialogs seem to be in the dark grey "dark room" mode):

![image](https://user-images.githubusercontent.com/3529789/67140300-8c37f780-f259-11e9-9fa2-762bba8ce788.png)

As you can see, this is clearly not fit for "dark room" use case. It represents a day situation, but a user in a dark room will have adaptation issues unless "night mode" is switched on.

This same screen would become almost fine when advancing to night time:

![image](https://user-images.githubusercontent.com/3529789/67140317-a376e500-f259-11e9-9231-a380e3484a6c.png)

The date/time dialog might need to be toned down on a dark room situation, because the local contrast might be too big, but that is not a big issue.

Now, to illustrate the proposed workaround, I take the day image and enable "night mode":

![image](https://user-images.githubusercontent.com/3529789/67140357-fea8d780-f259-11e9-8b30-632001f81fce.png)

Now I can use a daylight image in a dark room: notice that the date/time dialog is barely but sufficiently readable (but still better readable than the current graph dialogs!).

So the solution is

1. provide a normal daylight mode for the graphs
1. ensure there is enough contrast in "night mode"
1. ensure the lines a thick enough (single-pixel lines are not acceptable, because in night mode, the eye needs contrast)
1. ask the user to switch on night mode

optionally
1. add a night mode to the graphs (a separate button?)
1. choose other colors, that work in daylight mode as well as in night mode... 

So the essence is that the graphs should be presented in daylight mode rather than in "dark room" mode, otherwise the curves are ureadable, which is a shame for these awesome tools.

thank you for your efforts; if I find time, I will try to tackle this issue.

_ps - I believe I have a reasonable sense of ergonomy. Calling this an **opinion** is hurting, because it denigrates a perfectly fine argument, and I therefore have a difficult time to accept this not quite rational discussion. Also, asking for a technical solution if the present situation is not good enough is not nice, because many users here cannot provide a technical solution, but they know that developers can if they want to, so the "won't fix" label is a **strictly personal** decision, because technically it is possible and somewhere in the future, someone will provide the fix, contradicting the label. _

**Comment by axd1967** [comment]
Sorry, that's kind, but as I wrote, I currently have no time and don't want to block others from working out the solution. Whoever feels ready to work on it can always request to be assigned.

**Comment by gzotti** [comment]
You see, over 200 open issues, and still real and important problems open (our very opinion, indeed). 

We flag things that we find are opinions that differ from the views and preferences of the team as "opinion", and things that we therefore don't bother to fix ourselves with "won't fix". This does not disallow anybody of doing it, though, it just says "we don't want to do this because we feel our time should be be used elsewhere".

Although everybody could "just" do it and send a pull request with a working solution, nobody has so far volunteered to solve this problem. I assigned you because you are the only one who has expressed interest in this topic and at least some willingness to solve it. Be it next week, next year, in 2033, or after your retirement. If anybody else beats you in time, so be it.

**Comment by axd1967** [comment]
corrected in 0.19.2 - thank you

- checkboxes in bookmarks area readable

**Comment by axd1967** [comment]
interesting: https://github.com/Stellarium/stellarium/pull/1395

**Comment by axd1967** [comment]
This is what we need.

![image](https://user-images.githubusercontent.com/3529789/107819318-402b9300-6d79-11eb-8155-fa219a7f8f79.png)
![image](https://user-images.githubusercontent.com/3529789/107819337-4ae62800-6d79-11eb-85d4-1483f32f0dd2.png)

**Comment by axd1967** [comment]
Similarly, the sun lines should be drawn in tints of yellow. black  - or some dark color, as well as dark red tints - on dark grey is a bad choice (ergonomically speaking):

![image](https://user-images.githubusercontent.com/3529789/116413745-2f4ad380-a838-11eb-8e25-5936f1b2af7a.png)

oddly, the sun lines are only slightly more visible in night mode...

![image](https://user-images.githubusercontent.com/3529789/116413833-4093e000-a838-11eb-9177-a7b1dd97bc63.png)

I didn't check sourcecode history, but maybe the choice of the dark grey scheme may have been made in a time where night mode was not yet available. when night mode was introduced, the dark grey scheme should have been changed to a normal day (pastel) mode.

**Comment by gzotti** [comment]
Night mode used to be a set of entirely different colours (many years ago). Now it's just a GLSL filter over the RGB output.

**Comment by alex-w** [comment]
@axd1967, sorry, but you 2 years say that Stellarium is bad, the GUI is ugly, the ergonomy is not exist, etc. We proposed to you creating a pastel theme (as you want) and code for switching the themes. I’ve said before that this theme is good for me and I will do nothing for changing it. What is the results? Nothing from you - no theme, no code, nothing! Seriuosly, I think you not need any changes in the GUI.

**Comment by axd1967** [comment]
Nobody obliges anyone to do anything.

I never stated anywhere that "Stellarium is bad", on the contrary. If you can point that out, I'm glad to review that. That's not the same as saying that there are issues, but we all know there exists no application without issues.

I never stated "ergonomy does not exist". That's not the same as saying that there are some issues with ergonomy.

All I say is that Qt allows to use themes, and indeed it requires effort that is for the moment unavailable. 

And be honest: it's not because the theme is good for someone, that it is good for everybody. It's not because almost nobody is complaining, that there is no issue. The reality is simply that this is a difficult issue with no immediate solution, we just have to work towards the solution, and the first step is to acknowledge that there is a small problem for some people. Which means this issue cannot be given a high priority for now. Which does not prevent from further documenting the issues.

Can you please stop being so negative? I already contributed to Stellarium in helping, in code, in issues, and do not deserve this kind of attitude. It looks like this is all just a big misunderstanding. 

I'm not creating issues to prove that Stellarium is bad, but to make it better: call it positive criticism. And for my own reasons I cannot participate more actively, I'm sorry for that.

You guys are doing an awful lot of good work in your free time, and a huge community enjoys your (free!!!!) product. Let's continue that way.

**Comment by axd1967** [comment]
This is what I'm busy trying to improve:
![image](https://user-images.githubusercontent.com/3529789/121970777-97696080-cd77-11eb-984d-f4758625c85e.png)

**Comment by freebrowser1** [comment]
Well, this low contrast issue is not resolved yet: Stellarium 0.22.1 (macOS, Linux) still has barely readable dialogs (black on dark gray)
![Screenshot 2022-05-02 at 08 51 05](https://user-images.githubusercontent.com/26625141/166195502-f4119d52-9bce-4837-ad70-10d8363d9470.png)
.

**Comment by alex-w** [comment]
Please check details for adding user CSS in Stellarium User Guide - https://github.com/Stellarium/stellarium-data/releases/download/guide/guide.pdf
